src/bin/wasmer.rs

```diff
@@ -397,7 +403,7 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
             .map_err(|e| format!("Can't convert from wast to wasm: {:?}", e))?;
     }
 
-    let compiler: Box<dyn Compiler> = match get_compiler_by_backend(options.backend) {
+    let compiler: Box<dyn Compiler> = match get_compiler_by_backend(options.backend, options) {
```

Comment:
Maybe only pass `options` and let `get_compiler_by_backend` pull `backend` out of `options` itself?

Reply:
repinging on this

Reply:
When tiering is enabled, we can be using a different `backend` than the one specified in `options`.